### PR TITLE
test(async): improve test coverage

### PR DESCRIPTION
**Pull request by mbhrznr**
working towards https://github.com/denoland/deno_std/issues/3713.
pushes code coverage of the `async` sub-module closer to 100%.

the two remaining files in the sub-module which aren't at 100% are:
- `mux_async_iterator.ts`: 
   resetting the errors via `this.#throws.length = 0` seems to be unreachable code.
   if we have a truthy value for `this.#throws.length` we would throw all of the errors, thus exiting `iterate`.
- `pool.ts`:
   the `catch` block as well as the error handling seems to be covered by tests already.

### CODE REVIEW

File: async/pool_test.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ Deno.test("pooledMap()", async function () {
   assert(diff < 3000);
 });
 
-Deno.test("pooledMap() handles errors", async () => {
+Deno.test("pooledMap() handles errors", async function () {
```
Comment (iuioiua) [inline async/pool_test.ts:25]: Nit: we generally use the arrow syntax for anonymous functions. Please revert. Ditto elsewhere.

File: async/delay_test.ts [hunk 2]
```diff
@@ -103,3 +105,35 @@ Deno.test("delay() handles already aborted signal", async function () {
...
+  using unrefTimer = stub(Deno, "unrefTimer", () => {
+    throw new ReferenceError();
+  });
+  await delay(100, { persistent: false });
+  assertSpyCalls(unrefTimer, 1);
+});
+
+Deno.test({
+  name: "delay() handles persistent option with error",
+  fn: async function () {
```
Comment (iuioiua) [inline async/delay_test.ts:125]:
```suggestion
  async fn() {
```
Nit

**Review by iuioiua — CHANGES_REQUESTED** [review]
Nicely done! I have just a couple of nits. Let's also remove the unreachable code from `mux_async_iterator.ts`.

File: async/pool_test.ts (hunk 1 above)
Comment (mbhrznr) [reply async/pool_test.ts]: understood! also took liberty to convert the other test cases in the async sub-module to make use of the arrow syntax as well.

**Review by iuioiua — APPROVED** [review]
LGTM! Good stuff. Thank you.

**Pull request merged**